docs: prepare blog post for release (#38)
owasp-modsecurity/website: content/blog/2024-09-02-owasp-modsecurity-new-versions.md

---
title: 'New versions - 2024 August'
date: '2024-09-01T17:30:00+02:00'
author: airween
---

> OWASP ModSecurity team is pleased to announce the release of versions [2.9.8](https://github.com/owasp-modsecurity/ModSecurity/releases/tag/v2.9.8) and [3.0.13](https://github.com/owasp-modsecurity/ModSecurity/releases/tag/v3.0.13). These versions both include a mixture of new features and bug fixes.

Reviewer: ```suggestion
The OWASP ModSecurity team is pleased to announce the release of versions [2.9.8](https://github.com/owasp-modsecurity/ModSecurity/releases/tag/v2.9.8) and [3.0.13](https://github.com/owasp-modsecurity/ModSecurity/releases/tag/v3.0.13). These versions both include a mixture of new features and bug fixes.
```

<!--more-->

For the complete list of changes, please take a look at the respective CHANGELOGs: [mod_security2](https://github.com/owasp-modsecurity/ModSecurity/blob/v2.9.8/CHANGES) and [libmodsecurity3](https://github.com/owasp-modsecurity/ModSecurity/blob/v3.0.13/CHANGES).

> For some of the more complex changes, you may want to read through the corresponding pull requests (linked below) to understand rationals and implementation details.

Reviewer: ```suggestion
For some of the more complex changes, you may want to read through the corresponding pull requests (linked below) to understand rationales and implementation details.
```

It's been a long time since the last releases, especially in case of v2.

#### Major changes in v2 (details below):

* added a CI workflow
* changed error log format - [#1](#v2-1---changed-error-log-format)
* added a new MULTIPART HEADER check - [#2](#v2-2---added-a-new-multipart-header-check)
* fixed many potential memory leaks and other potential memory handling problems

##### v2 #1 - changed error log format

> See [PR #3192](https://github.com/owasp-modsecurity/ModSecurity/pull/3192). Here are how look the old an the new format:

Reviewer: ```suggestion
See [PR #3192](https://github.com/owasp-modsecurity/ModSecurity/pull/3192). The old log format differs from the new one as follows:
```

old:
```
[Wed Aug 28 17:07:09.416861 2024] [security2:error] [pid 729352:tid 729355] [client ::1:55806] [client ::1] ModSecurity ...
```
new:
```
[Wed Aug 28 17:07:09.416861 2024] [security2:error] [pid 729352:tid 729355] [client ::1:55806] ModSecurity ...
```
As you can see the second `[client]` field was removed.

##### v2 #2 - added a new MULTIPART HEADER check

> See [PR #3226](https://github.com/owasp-modsecurity/ModSecurity/pull/3226). For multipart requests, the engine checks that the header cannot contain non-ascii characters. This is a bit similar like CRS's [PR #3796](https://github.com/coreruleset/coreruleset/pull/3796) but on the engine's level. For more details, see the related [blog post](https://coreruleset.org/20240829/crs-versions-4.6.0-and-3.3.6-have-been-released/).

Reviewer: ```suggestion
See [PR #3226](https://github.com/owasp-modsecurity/ModSecurity/pull/3226). For multipart requests, the engine checks that the header does not contain invalid characters. This is similar to CRS's [PR #3796](https://github.com/coreruleset/coreruleset/pull/3796) but on the engine level. For more details, see the related [blog post](https://coreruleset.org/20240829/crs-versions-4.6.0-and-3.3.6-have-been-released/).
```

##### Contributors:

> @3eka, @airween, @fzipi, @marcstern, @martinhsv, @Polynomial-C, @twouters, @zhaoshikui.

Reviewer: I didn't provide any code but I did participate in the discussions. I won't be offended, if you don't mention me, just saying that I think I contributed something as well ;)

Author: Oh, sorry, you are definitely right. I just used GH's API and `grep` to collect the authors from PR's. I'll mention you in a separated block.

Author: Please check 955fcf322 - do you think it's ok?

(See the [Special thanks](https://github.com/owasp-modsecurity/website/blob/955fcf3225ec2b6e667e22d3ea64fee4c9c871c7/content/blog/2024-09-02-owasp-modsecurity-new-versions.md#special-thanks-to) section.)

Reviewer: Cool, thanks!

#### Major changes in v3:

* added Windows port - [#1](#v3-1---added-windows-port)
* improved CI workflow
* removed unnecessary string copy operations, improved engine speed - several PR's
* fixed a bug in `@pm` operator - [#2](#v3-2---fixed-a-bug-in-pm-operator)
* extended the C/C++ API - [#3](#v3-3---extended-the-cc-api)

##### v3 #1 - added Windows port

> See [PR #3132](https://github.com/owasp-modsecurity/ModSecurity/pull/3132). Libmodsecurity3 is able to built on Windows too.

Reviewer: ```suggestion
See [PR #3132](https://github.com/owasp-modsecurity/ModSecurity/pull/3132). Libmodsecurity3 builds on Windows now.
```

##### v3 #2 - fixed a bug in `@pm` operator

> See [PR #3243](https://github.com/owasp-modsecurity/ModSecurity/pull/3243) and [PR #3233](https://github.com/owasp-modsecurity/ModSecurity/pull/3233). Fixed parsing of digits which were not quoted and thus not interpreted as ASCII characters (like the hexadecimal digits) but as binary values, eg `0` was interpreted as string terminator (`'\0'`) and not ascii `'0'` (`chr(48)`).

Reviewer: ```suggestion
See [PR #3243](https://github.com/owasp-modsecurity/ModSecurity/pull/3243) and [PR #3233](https://github.com/owasp-modsecurity/ModSecurity/pull/3233). Fixed parsing of digits which were not quoted and thus not interpreted as ASCII characters (like the hexadecimal digits) but as binary values, eg `0` was interpreted as string terminator (`'\0'`) and not ASCII `'0'` (`chr(48)`).
```
